src/Services/UserManager.vala

```diff
@@ -35,24 +35,23 @@ public enum UserState {
 }
 
 public class Session.Services.UserManager : Object {
+    public signal void close ();
+
+    public bool has_guest { get; private set; default = false; }
+    public Session.Widgets.UserListBox user_grid { get; private set; }
+    public Wingpanel.Widgets.Separator users_separator { get; construct; }
+
+    private const uint GUEST_USER_UID = 999;
     private const uint NOBODY_USER_UID = 65534;
     private const uint RESERVED_UID_RANGE_END = 1000;
 
-    public signal void close ();
-
     private const string DM_DBUS_ID = "org.freedesktop.DisplayManager";
     private const string LOGIN_IFACE = "org.freedesktop.login1";
     private const string LOGIN_PATH = "/org/freedesktop/login1";
 
-    private signal void delete_user (ObjectPath user_path);
     private Act.UserManager manager;
-    private List<Widgets.Userbox> userbox_list;
+    private Gee.HashMap<uint, Widgets.Userbox>? users;
```

Comment:
I think I'd prefer it if the HashMap was called `user_widgets` or `user_rows` or something instead. `users` feels like it might be a list of things returned from accountsservice and it's not necessarily clear it contains Gtk Widgets.